src/python/tests_extended/test_tensor_based.py

```diff
@@ -0,0 +1,853 @@
+        inferencing_data1 = np.array([1.0, float("NaN"), 4.0, 7.0, float("NaN")]).astype(np.double).reshape(5,1)
+        inferencing_data2 = np.array(["1", "", "Hello", "", "World"]).reshape(5,1)
+        result = sess.run(None, {"data1": inferencing_data1, "data2": inferencing_data2})
+
+        expectedData1 = np.array([1.0, 1.0, 4.0, 7.0, 1.0]).astype(np.double).reshape(5, 1)
+        expectedData2 = np.array(["1", "two", "Hello", "two", "World"]).reshape(5, 1)
+        
+        np.testing.assert_array_equal(result[0],expectedData1)
+        np.testing.assert_array_equal(result[1],expectedData2)
+
+    def test_tostring_numbers_wo_dft(self):
+        training_data = pd.DataFrame(data=dict(
+            f0=[4, 4, -1, 9],
+            f1=[5, 5, 3.1, -0.23],
+            f2=[6, 6.7, np.nan, np.nan]
+        )).astype({'f0': np.int32,
+                   'f1': np.float32,
+                   'f2': np.float64})
+
+        xf = ToString(columns={'f0.out': 'f0',
+                               'f1.out': 'f1',
+                               'f2.out': 'f2'})
+                               
+        sess = set_up_onnx_model(xf, training_data)
+
+        inferencing_f0 = np.array([4, 4, -1, 9]).astype(np.int32).reshape(4,1)
+        inferencing_f1 = np.array([5, 5, 3.1, -0.23]).astype(np.float32).reshape(4,1)
+        inferencing_f2 = np.array([6, 6.7, float("NaN"), float("NaN")]).astype(np.float64).reshape(4,1)
+ 
+        # Run your inference session with your model and your data
+        result = sess.run(None, {"f0": inferencing_f0, "f1": inferencing_f1, "f2": inferencing_f2})
+        
+        f0_output = np.array([['4'], ['4'], ['-1'], ['9']]).reshape(4, 1)
+        f1_output = np.array([['5.000000'], ['5.000000'], ['3.100000'], ['-0.230000']]).reshape(4, 1)
+        f2_output = np.array([['6.000000'], ['6.700000'], ['NaN'], ['NaN']]).reshape(4, 1)
+
+        np.testing.assert_array_equal(f0_output, result[3])
+        np.testing.assert_array_equal(f1_output, result[4])
+        np.testing.assert_array_equal(f2_output, result[5])
+
+
+
+    def test_tostring_other_types_wo_dft(self):
+        training_data = pd.DataFrame(data=dict(
+            f0=[True, False],
+            f1=[123.45, 135453984983490.5473]
+        )).astype({'f0': bool,
+                   'f1': np.double})
+
+        xf = ToString(columns={'f0.out': 'f0',
+                               'f1.out': 'f1'})
+                               
+        sess = set_up_onnx_model(xf, training_data)
+
+        inferencing_f0 = np.array([True, False]).astype(bool).reshape(2,1)
+        inferencing_f1 = np.array([123.45, 135453984983490.5473]).astype(np.double).reshape(2,1)
+ 
+        # Run your inference session with your model and your data
+        result = sess.run(None, {"f0": inferencing_f0, "f1": inferencing_f1})
+        
+        f0_output = np.array([['True'], ['False']]).reshape(2, 1)
+        f1_output = np.array([['123.450000'], ['135453984983490.546875']]).reshape(2, 1)
```

Comment:
>.546875 [](start = 63, length = 7)

numbers dont match?

Reply:
They don't match because of the precision. Comment is added for this test.